core/src/main/java/org/elasticsearch/common/lucene/uid/PerThreadIDVersionAndSeqNoLookup.java

```diff
@@ -65,9 +71,24 @@
     PerThreadIDVersionAndSeqNoLookup(LeafReader reader) throws IOException {
```

Comment:
I tried to make changes around those lines. Information is kept in static threadlocals so we can't really pass the information at construction time, but it should be at least better validated now.